### PR TITLE
[#269] Ability to set prefetch on client consumer (and also own flow control)

### PR DESCRIPTION
**Pull request by pellmann**
- created AbstractConsumer
- moved methods to AbstractConsumer
- added variants with prefetch to factories and HonoClient
- added flow() to Consumer
- added prefetch property to JMeter consumer

### CODE REVIEW

File: client/src/main/java/org/eclipse/hono/client/MessageConsumer.java [hunk 1]
```diff
@@ -30,4 +31,13 @@
...
     void close(Handler<AsyncResult<Void>> closeHandler);
+
+    /**
+     * Grants the given number of message credits to the sender.
+     *
+     * For use when created with 0 prefetch in consumer creation
+     *
+     * @param credits the credits to flow
+     */
+    void flow(int credits);
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/MessageConsumer.java]: IMHO this method should throw a runtime exception if the consumer has been configured for automatic flow control.

File: client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java [hunk 2]
```diff
@@ -349,6 +337,18 @@ public HonoClient createTelemetryConsumer(
             final String tenantId,
             final Consumer<Message> telemetryConsumer,
             final Handler<AsyncResult<MessageConsumer>> creationHandler) {
+        return createTelemetryConsumer(tenantId,AbstractHonoClient.DEFAULT_SENDER_CREDITS,telemetryConsumer,creationHandler);
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java]: whitespaces after commas

File: jmeter/src/main/java/org/eclipse/hono/jmeter/HonoReceiverSampler.java [hunk 3]
```diff
@@ -45,6 +46,14 @@ public void setUseSenderTime(final Boolean useSenderTime) {
         setProperty(USE_SENDER_TIME, useSenderTime);
     }
 
+    public String getPrefetch() {
+        return getPropertyAsString(PREFETCH);
+    }
+
+    public void setPrefetch(final String prefetch) {
+        setProperty(PREFETCH,prefetch);
```
Comment (sophokles73) [inline jmeter/src/main/java/org/eclipse/hono/jmeter/HonoReceiverSampler.java]: whitespace

**Comment by pellmann** [comment]
Made requested changes/merged.

**Comment by sophokles73** [comment]
@pellmann can you please squash and edit the commit message to contain a non-breaking, self sufficient single line abstract (first line)?

**Comment by pellmann** [comment]
Done.

**Pull request merged**